engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -1235,4 +1235,30 @@ function updateLastAction($guid, $posted = null) {
 			return false;
 		}
 	}
+
+	/**
+	 * Get a user by GUID even if the entity is hidden or disabled
+	 *
+	 * @param int $guid User GUID. Default is logged in user
+	 *
+	 * @return \ElggUser|false
+	 * @access private
+	 */
+	public function resolveUser($guid = 0) {
```

Comment:
Strawman: move the functionality to get(), add $subtype (= null) and $ignore_condition (= false). For this use we'd set $type to "user" and $ignore_condition to true.